clang_tidy/clang_tidy_test.bzl

```diff
@@ -0,0 +1,85 @@
+load("@bazel_tools//tools/build_defs/cc:action_names.bzl", "ACTION_NAMES")
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load("clang_tidy.bzl", "safe_flags", "check_valid_file_type", "toolchain_flags", "rule_sources")
+
+def _clang_tidy_rule_impl(ctx):
+    wrapper = ctx.attr.clang_tidy_wrapper.files.to_list()[0]
+    exe = ctx.attr.clang_tidy_executable
+    additional_deps = ctx.attr.clang_tidy_additional_deps
+    config = ctx.attr.clang_tidy_config.files.to_list()[0]
+
+    c_flags = safe_flags(toolchain_flags(ctx, ACTION_NAMES.c_compile)) + ["-xc"]
+    cxx_flags = safe_flags(toolchain_flags(ctx, ACTION_NAMES.cpp_compile)) + ["-xc++"]
+    flags = cxx_flags
+
+    # Declare symlinks
+    clang_tidy_config = ctx.actions.declare_file(config.basename)
+    clang_tidy = ctx.actions.declare_file("run_clang_tidy.sh")
+
+    args = []
+    srcs = rule_sources(ctx, ctx.attr)
+
+    # specify the output file - twice
+    outfile = ctx.actions.declare_file(
+        "bazel_clang_tidy_.clang-tidy.yaml",
```

Comment:
this file likely needs to be based on the rule name so you can have 2 of this rule in the same directory w/o this conflicting